src/vm/jitinterface.cpp

```diff
@@ -5219,10 +5219,22 @@ void CEEInfo::getCallInfo(
             }
         }
 
-        MethodDesc * directMethod = constrainedType.GetMethodTable()->TryResolveConstraintMethodApprox(
-            exactType, 
-            pMD, 
-            &fForceUseRuntimeLookup);
+        // In a try block because the interface method resolution might end up being
+        // ambiguous (diamond inheritance case of default interface methods).
+        MethodDesc * directMethod = nullptr;
+        EX_TRY
+        {
+            directMethod = constrainedType.GetMethodTable()->TryResolveConstraintMethodApprox(
+                exactType,
+                pMD,
+                &fForceUseRuntimeLookup);
+        }
+        EX_CATCH
+        {
```

Comment:
Can this be eating other exceptions now - not just the ambiguous interface ones?

I am wondering whether it would be better to handle this lower; and return NULL from TryResolveConstraintMethodApprox for the ambiguous interface methods.

Reply:
When I was moving the catch to `TryResolveConstraintMethodApprox` and looked at how it's implemented, it made me realize that the way the code is written (trying to resolve on a canonical version first, and only then resolving on the exact type), it's likely it will hit the exception even for cases that are not actually ambiguous (they're only ambiguous when looking at them as `__Canon`). The code will likely have to be rewritten to deal with that.

While trying to set up a repro for that, I hit two more bugs, so I need to deal with those first before I can come back to this.